granary/microformats2.py

```diff
@@ -173,12 +186,15 @@ def object_to_json(obj, ctx={}, trim_nulls=True):
       objs = obj.get('object', [])
       if not isinstance(objs, list):
         objs = [objs]
-      ret['properties'][prop] = ret['properties'][prop + '-of'] = \
-          [o.get('url') for o in objs]
+      ret['properties'][prop + '-of'] = ret['properties'][prop] = [
+        o['url'] if 'url' in o and len(o) == 1
```

Comment:
agreed, i like that intuition. `like-of`s, `repost-of`s, etc should definitely prefer that shorter representation whenever possible.